agent/connect/csr.go

```diff
@@ -47,11 +48,28 @@ func SigAlgoForKeyType(keyType string) x509.SignatureAlgorithm {
 // along with the PEM-encoded private key for this certificate.
 func CreateCSR(uri CertURI, privateKey crypto.Signer,
 	dnsNames []string, ipAddresses []net.IP, extensions ...pkix.Extension) (string, error) {
+
+	// Drop everything after the ':' from the name when constructing the DNS SANs.
+	uniqueNames := make(map[string]struct{})
+	formattedDNSNames := make([]string, 0)
+	for _, host := range dnsNames {
+		hostSegments := strings.Split(host, ":")
+		formattedHost := hostSegments[0]
+		if _, ok := uniqueNames[formattedHost]; ok {
+			continue
+		}
+
+		if len(hostSegments) >= 1 {
```

Comment:
This check seems to be ineffective in its current position, because we're already assuming that `hostSegments` has at least 1 element by accessing its 0th element without a guard.